src/streams/adapters/EventedStreamSource.ts

```diff
@@ -0,0 +1,50 @@
+import Evented from '../../Evented';
+import { Handle } from '../../interfaces';
+import on, { EventEmitter, EventTarget, ExtensionEvent } from '../../on';
+import Promise from '../../Promise';
+import { Source } from '../ReadableStream';
+import ReadableStreamController from '../ReadableStreamController';
+
+type EventTargetTypes = Evented | EventEmitter | EventTarget;
+type EventTypes = Array<ExtensionEvent | string> | ExtensionEvent | string;
+
+export default class EventedStreamSource implements Source<Event> {
+	protected _controller: ReadableStreamController<Event>;
+	protected _target: EventTargetTypes;
+	protected _events: Array<ExtensionEvent | string>;
```

Comment:
Stylistic preference I guess.

`EventTargetTypes` is used in multiple places, so worth creating a type alias for.
`EventTypes` is a long garbled mess, so worth creating a type alias for (to keep the code on line 17 somewhat nicer looking)

Line 14 is the only place this union type is used, and it's short enough that it looks good inline.